pandas/io/formats/format.py

```diff
@@ -1720,11 +1720,12 @@ def format_percentiles(
             raise ValueError("percentiles should all be in the interval [0,1]")
 
     percentiles = 100 * percentiles
+    percentiles_round_type = percentiles.round().astype(int)
 
-    int_idx = np.isclose(percentiles.astype(int), percentiles)
+    int_idx = np.isclose(percentiles_round_type, percentiles)
 
     if np.all(int_idx):
-        out = percentiles.astype(int).astype(str)
+        out = percentiles_round_type.astype(str)
```

Comment:
Does the example in the original issue hit this path or the one below? Just want to make sure we have 2 tests that didn't work before and hit this change and the one below to work correctly now

Reply:
Hi, Does the example in the original issue hit this path or the one below?  

a) The example from issue will hit this path.
```
pd.DataFrame(data=[1, 2, 3], columns=['value']).describe(percentiles=[0.29, 0.57, 0.58])
```
```
if np.all(int_idx):
        out = percentiles_round_type.astype(str)
        return [i + "%" for i in out]
```
b) The example below I own create will hit below path.
```
pd.DataFrame(data=[1, 2, 3], columns=['value']).describe(percentiles=[0.291, 0.57, 0.58])
```
```
out[int_idx] = percentiles[int_idx].round().astype(int).astype(str)
```

Reply:
Could you add the `pd.DataFrame(data=[1, 2, 3], columns=['value']).describe(percentiles=[0.29, 0.57, 0.58])` case as a unit test?

Reply:
Have add test